iis-wf/iis-wf-referenceextraction/src/main/java/eu/dnetlib/iis/wf/referenceextraction/patent/PatentMetadataRetrieverJob.java

```diff
@@ -0,0 +1,167 @@
+                
+                storeInOutput(results._1, results._2, generateReportEntries(sc, results._1, results._2), outputPaths,
+                        params.numberOfEmittedFiles);
+                
+            } catch (ServiceFacadeException e) {
+                throw new RuntimeException("unable to instantiate patent service facade!", e);
+            }
+        }
+    }
+
+    //------------------------ PRIVATE --------------------------
+
+    private static JavaRDD<ReportEntry> generateReportEntries(JavaSparkContext sparkContext, 
+            JavaRDD<DocumentText> processedEntities, JavaRDD<Fault> processedFaults) {
+        
+        Preconditions.checkNotNull(sparkContext, "sparkContext has not been set");
+        
+        ReportEntry processedEnttiesCounter = ReportEntryFactory.createCounterReportEntry(COUNTER_PROCESSED_TOTAL, processedEntities.count());
+        ReportEntry processedFaultsCounter = ReportEntryFactory.createCounterReportEntry(COUNTER_PROCESSED_FAULT, processedFaults.count());
+        
+        return sparkContext.parallelize(Lists.newArrayList(processedEnttiesCounter, processedFaultsCounter));
+    }
+    
+    private static void storeInOutput(JavaRDD<DocumentText> retrievedPatentMeta, 
+            JavaRDD<Fault> faults, JavaRDD<ReportEntry> reports, OutputPaths outputPaths, int numberOfEmittedFiles) {
+        avroSaver.saveJavaRDD(retrievedPatentMeta.coalesce(numberOfEmittedFiles), DocumentText.SCHEMA$, outputPaths.getResult());
+        avroSaver.saveJavaRDD(faults.coalesce(numberOfEmittedFiles), Fault.SCHEMA$, outputPaths.getFault());
+        avroSaver.saveJavaRDD(reports.coalesce(numberOfEmittedFiles), ReportEntry.SCHEMA$, outputPaths.getReport());
+    }
+    
+    private static Tuple2<JavaRDD<DocumentText>, JavaRDD<Fault>> obtainMetadata(JavaRDD<ImportedPatent> input, PatentServiceFacade patentServiceFacade) {
+        JavaPairRDD<CharSequence, ContentRetrieverResponse> obtainedIdToResponce = input
+                .mapToPair(x -> new Tuple2<CharSequence, ContentRetrieverResponse>(getId(x),
+                        obtainMetadata(x, patentServiceFacade)));
+        obtainedIdToResponce.persist(StorageLevel.DISK_ONLY());
+        return new Tuple2<>(
+                obtainedIdToResponce.map(e -> DocumentText.newBuilder().setId(e._1).setText(e._2.getContent()).build()),
+                obtainedIdToResponce.filter(e -> e._2.getException() != null).map(e -> FaultUtils.exceptionToFault(e._1, e._2.getException(), null)));
+    }
+        
+    private static ContentRetrieverResponse obtainMetadata(ImportedPatent patent, PatentServiceFacade patentServiceFacade) {
+        try {
+            return new ContentRetrieverResponse(patentServiceFacade.getPatentMetadata(patent));
+        } catch (Exception e) {
+            log.error("Failed to obtain patent metadata for patent: " + patent.getApplnNr(), e);
+            return new ContentRetrieverResponse(e);
```

Comment:
Hmm, I think the reason for this flat approach with both fields defined alongside each other was the fact we treat the `content` field as required (even when empty) down the processing stream.

I would consider refining this approach in different, separate task because it is being used in different contexts than the patent retrieval.